### PR TITLE
run first vault node in membrane mode

### PR DESCRIPTION
**Pull request by maqi**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/maidsafe_vault/101)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @chandraprakash

(maidsafe_highfive has picked a reviewer for you, use r? to override)

**Comment by maqi** [comment]
failing in  Travis is because the executable can not be found , please make sure the executable has been built before the test.
failing in OSX is probably due to the test has been executed in parallel with other tests.

**Comment by rossmuir** [comment]
ok I will have a look @maqi

**Comment by chandraprakash** [comment]
As we are not going for first node option with routing as discussed in the meeting, this PR will not be required.

File: src/main.rs [hunk 1]
```diff
@@ -67,10 +77,74 @@ impl Vault {
...
+  The optional <peer>... arg(s) are a list of peer endpoints (other running vaults).
+  If these are supplied, the node will try to connect to one of these in order to
+  join the network.  If no endpoints are supplied, the node will try to connect to
+  an existing network using Crust's discovery protocol.
+";
+
+#[derive(RustcDecodable, Debug)]
+struct Args {
+    arg_peer: Vec<PeerEndpoint>,
+    flag_node: bool,
```
Comment (chandraprakash) [inline src/main.rs]: --node option is not required.

**Comment by benjaminbollen** [comment]
do the tests not need to get updated as well, to have a first routing_node start as "RoutingNode::run_zero_node"?

**Comment by chandraprakash** [comment]
@benjaminbollen  Its already done here https://github.com/maidsafe/maidsafe_vault/pull/101/files#diff-639fbc4ef05b315af92b4d836c31b023R209

File: src/main.rs [hunk 2]
```diff
@@ -119,9 +205,17 @@ mod test {
...
-        for i in 0..num_of_nodes {
+        // the first vault must be run in zero_membrane mode
+        println!("---------- starting node 0 --------------");
+        processes.push(match Command::new("./target/debug/maidsafe_vault").arg("-f").stdout(Stdio::piped()).spawn() {
+                    Err(why) => panic!("couldn't spawn maidsafe_vault: {}", why.description()),
+                    Ok(process) => process,
+                });
+        thread::sleep_ms(1000);
+
+        for i in 1..num_of_nodes {
```
Comment (chandraprakash) [inline src/main.rs]: this means we are running n+1 nodes.

**Comment by benjaminbollen** [comment]
okay cheers, sorry missed that !

File: src/main.rs [hunk 3]
```diff
@@ -67,10 +77,74 @@ impl Vault {
...
+    flag_first: bool,
+    flag_help: bool,
+}
+
+#[derive(Debug)]
+enum PeerEndpoint {
+    Tcp(SocketAddr),
+}
+
+impl Decodable for PeerEndpoint {
```
Comment (chandraprakash) [inline src/main.rs]: This is not required I think. We already have this in crust here : https://github.com/maidsafe/crust/blob/master/src/transport.rs#L54

File: src/main.rs (hunk 1 above)
Comment (maqi) [inline src/main.rs]: I prefer to have an explicit node option instead of using a default none

File: src/main.rs (hunk 2 above)
Comment (maqi) [inline src/main.rs]: 1..num_of_nodes only runs (num_of_nodes - 1) nodes, so it is still num_of_nodes in total

**Pull request merged**